chore(docs): docs versioning

signorecello opened the pull request:
trying baby steps here: docs will always deploy on master, but versioning is managed locally and is manually PRed into the repo

also releasing alpha-testnet.2 docs here, because it has broken links in master (CI was disabled)

signorecello committed:
chore(docs): splitting docs versioning

signorecello committed:
chore(docs): making docs versioning manual, but building them on master

signorecello committed:
chore(docs): making docs versioning manual, but building them on master

signorecello committed:
chore(docs): making docs versioning manual, but building them on master

signorecello committed:
cleanup

signorecello commented on bootstrap.sh:
```diff
@@ -269,8 +269,7 @@ function release {
     playground
-    # Blocking release.
-    # docs
+    docs
```
this enables docs builds on the CI (it calls the `""|"fast"|"full"` command in the docs folder

signorecello replied:
this will make the docs build on master, via the `""|"fast"|"full"` command that it was already calling

signorecello committed:
bug

signorecello commented on bootstrap.sh:
```diff
@@ -332,9 +331,9 @@ case "$cmd" in
+  "docs-cut-version")
     build
-    docs/bootstrap.sh docs-release
+    docs/bootstrap.sh $cmd "$@"
```
added a new script below that for now is only being called locally but can be done on a separate CI. baby steps

signorecello commented on docs/.gitignore:
```diff
@@ -27,5 +27,3 @@ yarn-error.log*
 /docs/docs/protocol-specs/public-vm/gen
 test-results
-
-versions.json
```
versioning is no longer managed by the CI: we're supposed to change this file to control which versions are built and shown on the website

signorecello commented on docs/bootstrap.sh:
```diff
@@ -18,7 +18,7 @@ if semver check $REF_NAME; then
 fi
 
-function build_and_preview {
+function build_docs {
```
separated build from the rest, since we're also building docs on master, the only difference is where they get published and whether they get a GH comment or not

signorecello committed:
more cleanup

signorecello commented on docs/bootstrap.sh:
```diff
@@ -67,40 +70,90 @@ function release_preview {
+    echo "Starting docs versioning for $COMMIT_TAG"
+
+    # Stash any unrelated local changes before starting
+    echo "Stashing local changes"
```
stashing in case there are changes to other files in the repo we want to persist

signorecello commented on docs/bootstrap.sh:
```diff
@@ -67,40 +70,90 @@ function release_preview {
+    # Checkout the tag, discarding local changes from previous build artifacts
+    # Use --force to overwrite potentially modified tracked files from the build process
+    echo "Checking out tag $COMMIT_TAG..."
+    if ! git checkout --force "$COMMIT_TAG"; then
```
checks out the version we specificed in the workflow

signorecello commented on docs/bootstrap.sh:
```diff
@@ -67,40 +70,90 @@ function release_preview {
+    # Create the versioned docs for this tag
+    echo "Creating documentation version for $COMMIT_TAG..."
+    # Pass COMMIT_TAG env var specifically if needed by the command
+    if ! COMMIT_TAG=$COMMIT_TAG yarn docusaurus docs:version "$COMMIT_TAG"; then
```
here's the actual version cutting, need to set `COMMIT_TAG` because of the `#include_aztec_version` macro

signorecello commented on docs/bootstrap.sh:
```diff
@@ -67,40 +70,90 @@ function release_preview {
-    ;;
-  "docs-release")
-    release
+  "docs-cut-version")
```
it's own separate action, so we run this locally and we make the PR... again, perfect for a follow-up github actions to do on its own

signorecello committed:
adding instructions to the README

signorecello committed:
making next version be called next, for now

signorecello committed:
releasing alpha-testnet.2 docs here because of broken links in master

signorecello committed:
restoring builds on builds

signorecello committed:
wip

signorecello committed:
wip

signorecello committed:
wip

signorecello committed:
wip

signorecello committed:
restoring paths

signorecello committed:
wip

signorecello committed:
wip

signorecello committed:
some cleanup

signorecello committed:
removing dangling release-please action

signorecello committed:
bug

signorecello committed:
inverting the deploy logic to account for empty dist tag

signorecello committed:
different strategy using github actions

nventuro committed:
docs: update utility fn docs (#13310)

Replacement for
https://github.com/AztecProtocol/aztec-packages/pull/13248 to get it out
of the graphite stack.

This PR updates mentions of top-level unconstrained contract fns for the
new 'utility' term. Some instances may be missing, but I think we
covered the vast majority of it. I also updated language in some parts
were explanations were outdated, and tried to remove bits that conflated
Noir unconstrained functions and utility functions. We'll likely want to
expand on this given the apparent confusion.

---------

Co-authored-by: benesjan <janbenes1234@gmail.com>

critesjosh commented on docs/bootstrap.sh:
```diff
@@ -67,40 +70,69 @@ function release_preview {
   ""|"full"|"fast")
-    build_and_preview
+    build_docs
+    deploy
```
Will this deploy every time someone runs bootstrap?

Commit:
chore: replace relative paths to noir-protocol-circuits

Commit:
git_subrepo.sh: Fix parent in .gitrepo file.

Commit:
git subrepo push --branch=master noir-projects/aztec-nr

subrepo:
  subdir:   "noir-projects/aztec-nr"
  merged:   "7ab89b96ff"
upstream:
  origin:   "https://github.com/AztecProtocol/aztec-nr"
  branch:   "master"
  commit:   "7ab89b96ff"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

fcarreiro committed:
fix(avm): request paths for appendLeaves (#13389)

Please read [contributing guidelines](CONTRIBUTING.md) and remove this
line.

fcarreiro committed:
feat(avm): tree padding (#13394)

Folded creation of db type aliases into this PR.

signorecello committed:
wip

signorecello committed:
wip

signorecello committed:
Merge remote-tracking branch 'origin' into zkpedro/docs_vers

signorecello committed:
wip

signorecello committed:
trigger

signorecello committed:
trigger

signorecello committed:
trigger

signorecello committed:
wip

signorecello committed:
dummy
